compiler/src/iree/compiler/Codegen/Common/LinkTuningSpecsPass.cpp

```diff
@@ -61,6 +61,71 @@ static bool consumesInputOp(NamedSequenceOp op) {
   return false;
 }
 
+static bool hasConsumedArgument(transform::ForeachMatchOp op) {
+  Value operand = op->getOperand(0);
+  if (auto blockArg = mlir::dyn_cast<BlockArgument>(operand)) {
+    Operation *parentOp = blockArg.getOwner()->getParentOp();
+    if (auto namedSequenceOp =
+            mlir::dyn_cast<transform::NamedSequenceOp>(parentOp)) {
+      return namedSequenceOp.getArgAttr(blockArg.getArgNumber(),
+                                        kArgConsumedAttrName) != nullptr;
+    }
+  }
+  return false;
+}
+
+static std::string
+getUniqueSpecName(StringRef specName,
+                  llvm::StringMap<unsigned> &specNameCounts) {
+  unsigned specNameSeenCount = specNameCounts[specName];
+  std::string newSpecName = specName.str();
+
+  // Keep incrementing the count until a truly unique name is found.
+  while (specNameSeenCount > 0 && specNameCounts.count(newSpecName)) {
+    newSpecName = llvm::formatv("{}_{}", specName, specNameSeenCount).str();
+    ++specNameSeenCount;
```

Comment:
This seem wasteful: not only we keep looping, but each time we also construct a new string. We should be able to come up with a unique name on the first try.